### PR TITLE
Add find cca command and list cca command.

### PR DESCRIPTION
**Pull request by mrnewguy**

### CODE REVIEW

File: src/main/java/seedu/jarvis/logic/commands/cca/ListCcaCommand.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+     * If the command has no inverse execution, then calling {@code executeInverse}
+     * will be guaranteed to always throw a {@code CommandException}.
+     *
+     * @return Whether the command has an inverse execution.
+     */
+    @Override
+    public boolean hasInverseExecution() {
+        return HAS_INVERSE;
+    }
+
```
Comment (nishcafe) [inline src/main/java/seedu/jarvis/logic/commands/cca/ListCcaCommand.java]: maybe you should write jdocs for the execute() method

File: src/test/java/seedu/jarvis/model/cca/CcaListTest.java [hunk 2]
```diff
@@ -96,10 +96,29 @@ public void setCca_editedCcaHasDifferentIdentity_success() {
...
+        assertThrows(NullPointerException.class, () -> ccaList.removeCca(null));
+    }
+
+    @Test
+    public void remove_ccaDoesNotExist_throwsCcaNotFoundException() {
+        assertThrows(CcaNotFoundException.class, () -> ccaList.removeCca(CANOEING));
+    }
+
+    @Test
+    public void remove_existingCca_removesCca() {
```
Comment (nishcafe) [inline src/test/java/seedu/jarvis/model/cca/CcaListTest.java]: you should do another test that removes a cca in the middle so that you know you're removing the cca at the correct index - zero based and one based numbers are tricky

File: src/main/java/seedu/jarvis/logic/commands/cca/FindCcaCommand.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    @Override
+    public boolean hasInverseExecution() {
+        return HAS_INVERSE;
+    }
+
+    /**
+     * Finds all {@code Person} in CcaTracker that pass the {@code Predicate} of the command.
+     *
+     * @param model {@code Model} which the command should operate on.
+     * @return {@code CommandResult} of the number of {@code Person} matching the {@code Predicate}.
```
Comment (gohsnn) [inline src/main/java/seedu/jarvis/logic/commands/cca/FindCcaCommand.java]: i think you meant `{@code CCA}` instead of `Person`?

File: src/test/java/seedu/jarvis/model/cca/CcaTrackerTest.java [hunk 4]
```diff
@@ -2,15 +2,32 @@
...
 import org.junit.jupiter.api.Test;
 
 public class CcaTrackerTest {
 
-    private final CcaTracker ccaTracker = new CcaTracker();
+    private CcaTracker ccaTracker;
+
+    @BeforeEach
+    public void setUp() {
+        // TODO is this the right way of stubbing? Do I need to stub the internalCcaList too?
```
Comment (gohsnn) [inline src/test/java/seedu/jarvis/model/cca/CcaTrackerTest.java]: try to address all your TODOs before making a PR to the team repo :) 
Alternatively u can make a static inner class `CcaStub` that extends `Cca` in your `CcaTrackerTest`

File: src/main/java/seedu/jarvis/logic/commands/cca/ListCcaCommand.java (hunk 1 above)
Comment (mrnewguy) [reply src/main/java/seedu/jarvis/logic/commands/cca/ListCcaCommand.java]: ok noted

File: src/main/java/seedu/jarvis/logic/commands/cca/EditCcaCommand.java [hunk 5]
```diff
@@ -114,7 +114,7 @@ public CommandResult execute(Model model) throws CommandException {
         Cca createdEditedCca = createEditedCca(originalCca, editCcaDescriptor);
 
         // checks if edited cca does not conflict with another existing cca that is not the original cca.
-        if (!originalCca.isSameCca(createdEditedCca) && model.hasCca(createdEditedCca)) {
+        if (!originalCca.isSameCca(createdEditedCca) && model.contains(createdEditedCca)) {
```
[the remark this answers is not in the document]
Comment (mrnewguy) [reply src/main/java/seedu/jarvis/logic/commands/cca/EditCcaCommand.java]: good idea, will implement in the next PR.

File: src/test/java/seedu/jarvis/model/cca/CcaTrackerTest.java (hunk 4 above)
Comment (mrnewguy) [reply src/test/java/seedu/jarvis/model/cca/CcaTrackerTest.java]: ok

File: src/main/java/seedu/jarvis/logic/commands/cca/EditCcaCommand.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (ryanYtan) [reply src/main/java/seedu/jarvis/logic/commands/cca/EditCcaCommand.java]: +1 AB3 implements it as `hasPerson()` as well.

File: src/main/java/seedu/jarvis/model/cca/CcaList.java [hunk 6]
```diff
@@ -30,11 +34,30 @@ public CcaList(){
      *
      * @param internalCcaList
      */
-    public CcaList(ArrayList<Cca> internalCcaList) {
+    public CcaList(ObservableList<Cca> internalCcaList) {
```
Comment (ryanYtan) [inline src/main/java/seedu/jarvis/model/cca/CcaList.java]: AB3's current implementation does not contain any constructor for their `PersonList`. The constructor may be redundant.
